packages/clay-core/docs/treeview.mdx

```diff
@@ -0,0 +1,378 @@
+
+```jsx
+<Mouse>{(mouse) => <Cat mouse={mouse} />}</Mouse>
+```
+
+The main components of the TreeView are `<TreeView.Item />`, `<TreeView.Group />` and `<TreeView.ItemStack />` which abstracts all the necessary markup complexity and allows the granular composition to build different styles of a TreeView, for example, File Explorer, Users, Page Navigation, Page Structure...
+
+### Static
+
+```jsx
+<TreeView>
+	<TreeView.Item>
+		<TreeView.ItemStack>
+			<Sticker displayType="primary" shape="user-icon" size="sm">
+				NS
+			</Sticker>
+			nisi quis eleifend
+		</TreeView.ItemStack>
+		<TreeView.Group>
+			<TreeView.Item key="Victor Valle">
+				<TreeView.ItemStack>
+					<Sticker displayType="primary" shape="user-icon" size="sm">
+						FP
+					</Sticker>
+					fusce ut placerat
+				</TreeView.ItemStack>
+				<TreeView.Group>
+					<TreeView.Item key="susana-vázquez">
+						<Sticker
+							displayType="primary"
+							shape="user-icon"
+							size="sm"
+						>
+							UT
+						</Sticker>
+						ultrices dui sapien
+					</TreeView.Item>
+				</TreeView.Group>
+			</TreeView.Item>
+			<TreeView.Item key="emily-young">
+				<Sticker displayType="primary" shape="user-icon" size="sm">
+					MC
+				</Sticker>
+				maecenas pharetra convallis
+			</TreeView.Item>
+		</TreeView.Group>
+	</TreeView.Item>
+</TreeView>
+```
+
+### Dynamic
+
+The TreeView and Group components when used with dynamic items are populated from hierarchical data. The component consumes the data using the `items` property.
+
+For the component to be dynamic it is necessary to convert `children` to Render prop, which must be a function that will receive the `item` of the current iterator.
+
+```jsx
+function Example() {
+	const items = [
+		{
+			children: [
+				{
+					children: [{name: 'ultrices dui sapien'}],
+					name: 'fusce ut placerat',
+				},
+				{name: 'maecenas pharetra convallis'},
+			],
+			name: 'nisi quis eleifend',
+		},
+	];
+
+	return (
+		<TreeView items={items} nestedKey="children">
+			{(item) => (
+				<TreeView.Item key={item.name}>
+					{item.name}
+
+					{item.children && (
+						<TreeView.Group items={item.children}>
+							{(item) => (
+								<TreeView.Item key={item.name}>
+									{item.name}
+								</TreeView.Item>
+							)}
+						</TreeView.Group>
+					)}
+				</TreeView.Item>
+			)}
+		</TreeView>
+	);
+}
+```
+
+#### Asynchronous Item
+
+Asynchronous Item loading is quite advantageous to decrease the initial payload and decrease memory space, this is more efficient when a tree is very large.
+
+The TreeView doesn't know when an Item is asynchronous so a deduction is made that the developer must deal with on the implementation side to tell whether the Item is asynchronous or not. The `onLoadMore` property is called every time the Item is a leaf of the tree and depending on the method's return it will make some assumptions.
```

Comment:
This got better, I'll still keep the leaf, this normally refers to when node is the last level of the tree because actually, we will only call the method when it is the last level.

Reply:
No problem. I was thinking that using too many terms: tree, node, leaf might lead to some confusion, if we keep on like that we might end up using "fruit" :joy: But I'm OK with using leaf for now